samwise.js

```diff
@@ -1,39 +1,42 @@
 /**
```

Comment:
SCRIPT-This new feature lacks unit tests. Could you add some to ensure everything works correctly?